### PR TITLE
Master codacy check

### PR DESCRIPTION
**Pull request by cqy123456**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

**Which branch you want to cherry-pick to?**

Not Available

**Which issue(s) this PR fixes:**

Fixes #

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

File: core/src/db/IDGenerator.cpp [hunk 1]
```diff
@@ -12,8 +12,8 @@
 #include "db/IDGenerator.h"
 #include "utils/Log.h"
 
-#include <assert.h>
```
Comment (cydrain) [inline core/src/db/IDGenerator.cpp:15]: try remove it

File: core/src/index/knowhere/knowhere/index/vector_index/IndexAnnoy.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ IndexAnnoy::Serialize(const Config& config) {
 
     auto index_length = index_->get_index_length();
```
Comment (cydrain) [inline core/src/index/knowhere/knowhere/index/vector_index/IndexAnnoy.cpp:43]: you'd better change here:
size_t index_length = index_->get_index_length();

File: core/src/index/knowhere/knowhere/index/vector_index/IndexAnnoy.cpp [hunk 3]
```diff
@@ -98,7 +98,7 @@ IndexAnnoy::BuildAll(const DatasetPtr& dataset_ptr, const Config& config) {
     }
 
     for (int i = 0; i < rows; ++i) {
-        index_->add_item(p_ids[i], (const float*)p_data + dim * i);
+        index_->add_item(p_ids[i], static_cast<const float*>(p_data) + dim * i);
```
Comment (cydrain) [inline core/src/index/knowhere/knowhere/index/vector_index/IndexAnnoy.cpp:101]: use reinterpret_cast

File: core/src/index/knowhere/knowhere/index/vector_index/IndexBinaryIVF.cpp [hunk 4]
```diff
@@ -132,8 +133,8 @@ BinaryIVF::QueryImpl(int64_t n, const uint8_t* data, int64_t k, float* distances
     ivf_index->nprobe = params->nprobe;
 
     stdclock::time_point before = stdclock::now();
-    int32_t* i_distances = reinterpret_cast<int32_t*>(distances);
-    index_->search(n, (uint8_t*)data, k, i_distances, labels, bitset_);
+    auto* i_distances = reinterpret_cast<int32_t*>(distances);
```
Comment (cydrain) [inline core/src/index/knowhere/knowhere/index/vector_index/IndexBinaryIVF.cpp:136]: auto i_distances = reinterpret_cast<int32_t*>(distances);

File: core/src/index/knowhere/knowhere/index/vector_index/gpu/IndexGPUIDMAP.h [hunk 5]
```diff
@@ -45,6 +45,8 @@ class GPUIDMAP : public IDMAP, public GPUIndex {
     void
     GenGraph(const float*, const int64_t, GraphType&, const Config&);
 
+    virtual ~GPUIDMAP(){};
```
Comment (cydrain) [inline core/src/index/knowhere/knowhere/index/vector_index/gpu/IndexGPUIDMAP.h:48]: virtual ~GPUIDMAP() = default;

File: core/src/index/knowhere/knowhere/index/vector_offset_index/IndexIVF_NM.cpp [hunk 6]
```diff
@@ -77,27 +77,26 @@ IVF_NM::Load(const BinarySet& binary_set) {
...
 #else
     auto rol = dynamic_cast<faiss::ReadOnlyArrayInvertedLists*>(invlists);
     auto lengths = rol->readonly_length;
-    auto rol_ids = (const int64_t*)rol->pin_readonly_ids->data;
+    auto rol_ids = reinterpret_cast<const int64_t*>(rol->pin_readonly_ids->data);
     for (size_t i = 0; i < invlists->nlist; i++) {
         auto list_size = lengths[i];
         for (size_t j = 0; j < list_size; j++) {
-            memcpy(arranged_data + d * sizeof(float) * (curr_index + j), original_data + d * rol_ids[curr_index + j],
+            memcpy(arranged_data + d * (curr_index + j), original_data + d * rol_ids[curr_index + j],
```
Comment (cydrain) [inline core/src/index/knowhere/knowhere/index/vector_offset_index/IndexIVF_NM.cpp:92]: need check

**Review by cydrain — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]

**Pull request merged**